zenodo/modules/metrics/api.py

```diff
@@ -0,0 +1,76 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Zenodo.
+# Copyright (C) 2016, 2017 CERN.
+#
+# Zenodo is free software; you can redistribute it
+# and/or modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# Zenodo is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Zenodo; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+# MA 02111-1307, USA.
+#
+# In applying this license, CERN does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+"""Zenodo Communities API."""
+
+from __future__ import absolute_import
+
+from invenio_db import db
+from invenio_pidrelations.contrib.versioning import PIDVersioning
+from invenio_pidstore.models import PersistentIdentifier
+
+
+class ZenodoMetric(object):
+    """API class for Zenodo Metrics."""
+
+    @staticmethod
+    def get_users(cache_timeout=-1):
```

Comment:
```suggestion
    def get_users():
```

I would say the caching can be done on a higher level (i.e. in the `utils.py` file).

Reply:
The way I envisioned this was that `get_users` would check if there was a cached value and return that or compute it and save it in the cache with `cache_timeout`. This would get repetitive for all functions, but we would eliminate the need for a separate config key.

Works either way for me.